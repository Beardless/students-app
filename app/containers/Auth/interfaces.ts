import { ModuleInterface } from "../SingleCourse/interfaces";

export interface AuthorInterface {
  bio: string;
  href: {
    picture: string;
  };
  name: string;
  picture: string;
}

interface AuthFields {
  type: string;
  label: string;
  value: string;
  disabled: boolean;
  error: string;
}

export interface CourseInterface {
  id?: number;
  title: string;
  description: string;
  cover_image_name: string;
  created: string;
  updated: string;
  href: {
    cover_image_url: string;
  };
  modules?: ModuleInterface[];
  authors?: AuthorInterface[];
}

export interface UserInterface {
  email: string;
  created: string;
  first_name: string;
  last_name: string;
  id: string;
  updated: string;
  roles: string[];
  courses: CourseInterface[];
}

export interface AuthFieldsInterface extends Array<AuthFields> {}
