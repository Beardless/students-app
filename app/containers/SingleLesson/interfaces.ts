export interface LessonInterface {
  id: string;
  title: string;
  description: string;
  position: number;
  embed_code: string;
  created: string;
  updated: string;
  module: {
    id: string;
    title: string;
    course: {
      id: number;
      title: string;
    };
  };
}
