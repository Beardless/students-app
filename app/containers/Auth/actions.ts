import {
  AUTH_ACTION,
  AUTH_ACTION_ERROR,
  AUTH_ACTION_SUCCESS,
  userLoginInterface
} from "./constants";

export function authAction(
  payload: userLoginInterface
): { type: string; payload: userLoginInterface } {
  return {
    type: AUTH_ACTION,
    payload
  };
}

export function authActionSuccess(user: any): { type: string; user: any } {
  return {
    type: AUTH_ACTION_SUCCESS,
    user
  };
}

export function authActionError(
  error: string
): { type: string; error: string } {
  return {
    type: AUTH_ACTION_ERROR,
    error
  };
}
