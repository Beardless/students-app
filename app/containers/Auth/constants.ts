export const AUTH_ACTION: string = "courses/Auth/AUTH_ACTION";
export const AUTH_ACTION_SUCCESS: string = "courses/Auth/AUTH_ACTION_SUCCESS";
export const AUTH_ACTION_ERROR: string = "courses/Auth/AUTH_ACTION_ERROR";
export const USER_ACTION: string = "courses/Auth/USER_ACTION";

export interface userLoginInterface {
  username: string;
  password: string;
}
