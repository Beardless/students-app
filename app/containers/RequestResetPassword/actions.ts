import {
  REQUEST_RESET_PASSWORD,
  REQUEST_RESET_PASSWORD_ERROR,
  REQUEST_RESET_PASSWORD_SUCCESS
} from "./constants";

export function requestResetPasswordAction(
  payload: string
): { type: string; payload: string } {
  return {
    type: REQUEST_RESET_PASSWORD,
    payload
  };
}

export function requestResetPasswordActionSuccess(
  email: string
): { type: string; payload: string } {
  return {
    type: REQUEST_RESET_PASSWORD_SUCCESS,
    payload: email
  };
}

export function requestResetPasswordActionError(
  error: string
): { type: string; error: string } {
  return {
    type: REQUEST_RESET_PASSWORD_ERROR,
    error
  };
}
