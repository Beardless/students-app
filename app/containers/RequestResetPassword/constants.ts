export const REQUEST_RESET_PASSWORD: string =
  "courses/RequestResetPassword/REQUEST_RESET_PASSWORD";
export const REQUEST_RESET_PASSWORD_SUCCESS: string =
  "courses/RequestResetPassword/REQUEST_RESET_PASSWORD_SUCCESS";
export const REQUEST_RESET_PASSWORD_ERROR: string =
  "courses/RequestResetPassword/REQUEST_RESET_PASSWORD_ERROR";
