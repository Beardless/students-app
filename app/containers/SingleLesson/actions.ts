import {
  LESSON_ACTION,
  LESSON_ACTION_SUCCESS,
  LESSON_ACTION_ERROR,
  LESSON_COMPLETE_ACTION,
  LESSON_COMPLETE_ACTION_SUCCESS,
  LESSON_COMPLETE_ACTION_ERROR
} from "./constants";
import { LessonInterface } from "./interfaces";

export function lessonAction(
  id: string
): { type: string; payload: { id: string } } {
  return {
    type: LESSON_ACTION,
    payload: { id }
  };
}

export function lessonActionSuccess(
  lesson: LessonInterface
): { type: string; lesson: LessonInterface } {
  return {
    type: LESSON_ACTION_SUCCESS,
    lesson
  };
}

export function lessonActionError(
  error: string
): { type: string; error: string } {
  return {
    type: LESSON_ACTION_ERROR,
    error
  };
}

export function lessonCompleteAction(
  isComplete: boolean,
  lessonId: string
): { type: string; payload: { isComplete: boolean; lessonId: string } } {
  return {
    type: LESSON_COMPLETE_ACTION,
    payload: {
      isComplete,
      lessonId
    }
  };
}

export function lessonCompleteActionSuccess(
  lesson: LessonInterface
): { type: string; lesson: LessonInterface } {
  return {
    type: LESSON_COMPLETE_ACTION_SUCCESS,
    lesson
  };
}

export function lessonCompleteActionError(
  error: string
): { type: string; error: string } {
  return {
    type: LESSON_COMPLETE_ACTION_ERROR,
    error
  };
}
