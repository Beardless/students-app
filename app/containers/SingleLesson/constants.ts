//Lesson action
export const LESSON_ACTION: string = "courses/SingleLesson/LESSON_ACTION";
export const LESSON_ACTION_SUCCESS: string =
  "courses/SingleLesson/LESSON_ACTION_SUCCESS";
export const LESSON_ACTION_ERROR: string =
  "courses/SingleLesson/LESSON_ACTION_ERROR";

//Lesson complete action
export const LESSON_COMPLETE_ACTION: string =
  "courses/SingleLesson/LESSON_COMPLETE_ACTION";
export const LESSON_COMPLETE_ACTION_SUCCESS: string =
  "courses/SingleLesson/LESSON_COMPLETE_ACTION_SUCCESS";
export const LESSON_COMPLETE_ACTION_ERROR: string =
  "courses/SingleLesson/LESSON_COMPLETE_ACTION_ERROR";

//Create bookmark action
export const CREATE_BOOKMARK_ACTION: string =
  "courses/SingleLesson/CREATE_BOOKMARK_ACTION";
export const CREATE_BOOKMARK_ACTION_SUCCESS: string =
  "courses/SingleLesson/CREATE_BOOKMARK_ACTION_SUCCESS";
export const CREATE_BOOKMARK_ACTION_ERROR: string =
  "courses/SingleLesson/CREATE_BOOKMARK_ACTION_ERROR";

//Get bookmark list action
export const GET_BOOKMARK_LIST_ACTION: string =
  "courses/SingleLesson/GET_BOOKMARK_LIST_ACTION";
export const GET_BOOKMARK_LIST_ACTION_SUCCESS: string =
  "courses/SingleLesson/GET_BOOKMARK_LIST_ACTION_SUCCESS";
export const GET_BOOKMARK_LIST_ACTION_ERROR: string =
  "courses/SingleLesson/GET_BOOKMARK_LIST_ACTION_ERROR";

//Delete bookmark action
export const DELETE_BOOKMARK_ACTION: string =
  "courses/SingleLesson/DELETE_BOOKMARK_ACTION";
export const DELETE_BOOKMARK_ACTION_SUCCESS: string =
  "courses/SingleLesson/DELETE_BOOKMARK_ACTION_SUCCESS";
export const DELETE_BOOKMARK_ACTION_ERROR: string =
  "courses/SingleLesson/DELETE_BOOKMARK_ACTION_ERROR";
