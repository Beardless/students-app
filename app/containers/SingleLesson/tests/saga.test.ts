import getSingleLesson, { getLesson } from "../saga";
import * as lesson from "./mocks/lesson.json";
import { put, takeEvery } from "redux-saga/effects";
import { lessonActionError, lessonActionSuccess } from "../actions";
import { LESSON_ACTION } from "../constants";
import { LessonInterface } from "../interfaces";

const lessonPayload = { payload: { id: "1" } };

describe("getLesson Saga", () => {
  let getLessonGenerator;

  beforeEach(() => {
    getLessonGenerator = getLesson(lessonPayload);
    const callDescriptor = getLessonGenerator.next(lessonPayload).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it("should dispatch the lessonActionSuccess if it requests the data successfully", () => {
    const response: { data: LessonInterface } = { data: lesson };
    const putDescriptor = getLessonGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(lessonActionSuccess(response.data)));
  });

  it("should call the authActionError if the response errors", () => {
    const err = { response: { data: { message: "Test error" } } };
    const putDescriptor = getLessonGenerator.throw(err).value;
    expect(putDescriptor).toEqual(put(lessonActionError("Test error")));
  });
});

describe("getSingleLesson Saga", () => {
  const getSingleLessonSaga = getSingleLesson();

  it("should start task to watch for LESSON_ACTION action", () => {
    const takeEveryDescriptor = getSingleLessonSaga.next().value;
    // @ts-ignore
    expect(takeEveryDescriptor).toEqual(takeEvery(LESSON_ACTION, getLesson));
  });
});
