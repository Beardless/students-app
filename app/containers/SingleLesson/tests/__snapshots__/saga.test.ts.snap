// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLesson Saga should call the authActionError if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "GET",
        "url": "/lessons/1",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getLesson Saga should dispatch the lessonActionSuccess if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "GET",
        "url": "/lessons/1",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
